src/Streamly/Internal/Data/SmallArray.hs

```diff
@@ -50,7 +50,7 @@ import Control.Monad (when)
 import Control.Monad.IO.Class (MonadIO, liftIO)
 import GHC.IO (unsafePerformIO)
 import Data.Functor.Identity (runIdentity)
-import Data.Primitive.SmallArray
+import Streamly.Internal.Data.SmallArray.Types
```

Comment:
```suggestion

import Streamly.Internal.Data.SmallArray.Types
```

Reply:
We can import from Data.Primitive.SmallArray if we are using a GHC / primitive package that comes with SmallArray and save on compilation time.

Reply:
That is what I have written in the beginning but as @harendra-kumar mentioned, debugging arrays in the future might prove difficult among other problems.

Reply:
No point in compiling everything twice!

Reply:
We might want to rename `SmallArray` to `Array` in the future. We will not be using `Data.Primitive.SmallArray` even if it is available.